src/VisualStudio/Core/Def/Implementation/TaskList/ProjectExternalErrorReporter.cs

```diff
@@ -224,10 +227,16 @@ private static DiagnosticSeverity GetDiagnosticSeverity(ExternalError error)
             string mappedFilePath, int mappedStartLine, int mappedStartColumn, int mappedEndLine, int mappedEndColumn,
             string originalFilePath, int originalStartLine, int originalStartColumn, int originalEndLine, int originalEndColumn)
         {
+            var document = id != null ? _workspace.CurrentSolution.GetDocument(id) : null;
+            var sourceSpan = document != null ?
+                document.GetSpanFromPositionAsync(originalStartLine, originalStartColumn, originalEndLine, originalEndColumn, CancellationToken.None).WaitAndGetResult(CancellationToken.None) :
```

Comment:
/cc @srivatsn 

`are we supporting code action (including suppression) for build error? there is no way to know the build error is valid or not unless you build before running LB like FxCop` - Ideally, we would like to but as you mention we won't have any way to detect if it is stale or not. Also suppressing a build diagnostic from error list won't update its suppression state as we won't recompute it - so probably we should just show a blank suppression state for build diagnostics?

Thanks for pointing me to the right method to get text span - I will revert the change in this file, and compute the span in the suppression fix service when we are required to fix build diagnostics.